engine/src/cmd/unit_generic.cpp

```diff
@@ -5930,7 +5930,7 @@ Vector Unit::MountPercentOperational( int whichmount )
     return Vector( mounts[whichmount].functionality,
                   mounts[whichmount].maxfunctionality,
                   ( (mounts[whichmount].status == Mount::ACTIVE || mounts[whichmount].status
-                     == Mount::INACTIVE) ? 0.0 : (Mount::UNCHOSEN ? 2.0 : 1.0) ) );
+                     == Mount::INACTIVE) ? 0.0 : (mounts[whichmount].status == Mount::UNCHOSEN ? 2.0 : 1.0) ) );
```

Comment:
This looks correct. Good job.

Reply:
Heh, it was quite the adventure getting there. I first had to figure out what the "?" was. Then, I pulled it apart, and made a proper if statement from it. When I was still getting the compiler error, it finally dawned on me that one of the three "boolean" comparisons were not like the other. 
So, this inline if is just one step short of a switch statement?